src/System.Runtime/tests/System/ActivatorTest.netstandard1.7.cs

```diff
@@ -0,0 +1,101 @@
+            Activator.CreateInstance(typeof(ClassWithPrivateCtor), BindingFlags.Static | BindingFlags.NonPublic, null, null, CultureInfo.CurrentCulture);
+            Assert.Equal(300, Flag.cnt);
+            Activator.CreateInstance(typeof(ClassWithPrivateCtor), BindingFlags.Instance | BindingFlags.NonPublic, null, null, CultureInfo.CurrentCulture);
+            Assert.Equal(500, Flag.cnt);
+
+            Flag.Reset(0); Assert.Equal(0, Flag.cnt);
+            Activator.CreateInstance(typeof(ClassWithPrivateCtor2), BindingFlags.Instance | BindingFlags.NonPublic, null, null, CultureInfo.CurrentCulture);
+            Assert.Equal(300, Flag.cnt);
+            Activator.CreateInstance(typeof(ClassWithPrivateCtor2), BindingFlags.Static | BindingFlags.NonPublic, null, null, CultureInfo.CurrentCulture);
+            Assert.Equal(500, Flag.cnt);
+
+            Flag.Reset(0); Assert.Equal(0, Flag.cnt);
+            Activator.CreateInstance(typeof(ClassWithPrivateCtor3), BindingFlags.Instance | BindingFlags.Public, null, new object[] { 122 }, CultureInfo.CurrentCulture);
+            Assert.Equal(400, Flag.cnt);
+            Activator.CreateInstance(typeof(ClassWithPrivateCtor3), BindingFlags.Static | BindingFlags.NonPublic, null, null, CultureInfo.CurrentCulture);
+            Assert.Equal(600, Flag.cnt);
+            Activator.CreateInstance(typeof(ClassWithPrivateCtor3), BindingFlags.Instance | BindingFlags.Public, null, new object[] { 122 }, CultureInfo.CurrentCulture);
+            Assert.Equal(900, Flag.cnt);
+
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateCtor), BindingFlags.Public | BindingFlags.Static, null, null, CultureInfo.CurrentCulture));
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateCtor), BindingFlags.Public | BindingFlags.Static, null, new object[] { 122 }, CultureInfo.CurrentCulture));
+        }
+
+        [Fact]
+        static void TestingBindingFlags1()
+        {
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateCtor), BindingFlags.Public | BindingFlags.Instance, null, null, CultureInfo.CurrentCulture, null));
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateCtor), BindingFlags.Public | BindingFlags.Instance, null, new object[] { 1, 2, 3 }, CultureInfo.CurrentCulture, null));
+
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateCtor), BindingFlags.Public | BindingFlags.Static, null, null, CultureInfo.CurrentCulture, null));
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateCtor), BindingFlags.Public | BindingFlags.Static, null, new object[] { 122 }, CultureInfo.CurrentCulture, null));
+        }
+
+        [Fact]
+        static void TestingActivationAttributes()
```

Comment:
> If that is the case we should remove the overloads that have ActivationAttributes, as no valid ActivationAttributes can ever be passed, makes sense?

There are quite a few methods in netstandard that do not make a lot of sense. This is just yet another one.

In CoreCLR, we will always take this path https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/Activator.cs#L97. If you call 

`Activator.CreateInstance(typeof(FileStream), ....., new Object[] { new Object() })`, it will throw NotSupportedException that says "Activation Attributes are not supported for types not deriving from MarshalByRefObject". It is bogus error message because of you have asked it to create FileStream that does derive from MarshalByRefObject. I am saying that it would be more appropriate to just throw PNSE exception with "Activation Attributes are not supported" message.

> that remoting could potentially work on top 

The question about which exception to throw for not-supported behavior has nothing to do with whether the remoting public surface is extension to netstandard public surface.